examples/src/main/kotlin/examples/cnn/mnist/advanced/LeNetWithTimeStoppingCallback.kt

```diff
@@ -0,0 +1,129 @@
+
+private const val EPOCHS = 10
+private const val TRAINING_BATCH_SIZE = 1000
+private const val NUM_CHANNELS = 1L
+private const val IMAGE_SIZE = 28L
+private const val SEED = 12L
+private const val TEST_BATCH_SIZE = 1000
+
+/**
+ * This is an CNN based on an implementation of LeNet-5 from classic paper trained with TimeStopping callback.
+ *
+ * @see <a href="http://yann.lecun.com/exdb/publis/pdf/lecun-98.pdf">
+ *    Gradient-based learning applied to document recognition:[Yann LeCun, Léon Bottou, Yoshua Bengio, Patrick Haffner, 1998]</a>
+ */
+private val lenet5Classic = Sequential.of(
+    Input(
+        IMAGE_SIZE,
+        IMAGE_SIZE,
+        NUM_CHANNELS
+    ),
+    Conv2D(
+        filters = 6,
+        kernelSize = longArrayOf(5, 5),
+        strides = longArrayOf(1, 1, 1, 1),
+        activation = Activations.Tanh,
+        kernelInitializer = GlorotNormal(SEED),
+        biasInitializer = Zeros(),
+        padding = ConvPadding.SAME
+    ),
+    AvgPool2D(
+        poolSize = intArrayOf(1, 2, 2, 1),
+        strides = intArrayOf(1, 2, 2, 1),
+        padding = ConvPadding.VALID
+    ),
+    Conv2D(
+        filters = 16,
+        kernelSize = longArrayOf(5, 5),
+        strides = longArrayOf(1, 1, 1, 1),
+        activation = Activations.Tanh,
+        kernelInitializer = GlorotNormal(SEED),
+        biasInitializer = Zeros(),
+        padding = ConvPadding.SAME
+    ),
+    AvgPool2D(
+        poolSize = intArrayOf(1, 2, 2, 1),
+        strides = intArrayOf(1, 2, 2, 1),
+        padding = ConvPadding.VALID
+    ),
+    Flatten(), // 3136
+    Dense(
+        outputSize = 120,
+        activation = Activations.Tanh,
+        kernelInitializer = GlorotNormal(SEED),
+        biasInitializer = Constant(0.1f)
+    ),
+    Dense(
+        outputSize = 84,
+        activation = Activations.Tanh,
+        kernelInitializer = GlorotNormal(SEED),
+        biasInitializer = Constant(0.1f)
+    ),
+    Dense(
+        outputSize = NUMBER_OF_CLASSES,
+        activation = Activations.Linear,
+        kernelInitializer = GlorotNormal(SEED),
+        biasInitializer = Constant(0.1f)
+    )
+)
+
+/**
+ * This example shows how to do image classification from scratch using [lenet5Classic], without leveraging pre-trained weights or a pre-made model.
+ * We demonstrate the workflow on the Mnist classification dataset.
+ *
+ * It includes:
+ * - dataset loading from S3
+ * - callback definition
+ * - model compilation with [TimeStopping] callback
+ * - model summary
+ * - model training
+ * - model evaluation
+ */
+fun lenetWithTimeStoppingCallback() {
+    val (train, test) = mnist()
+
+    lenet5Classic.use {
+        val timeStopping = TimeStopping(seconds = 30, verbose = true)
+        it.compile(
+            optimizer = Adam(clipGradient = ClipGradientByValue(0.1f)),
+            loss = Losses.SOFT_MAX_CROSS_ENTROPY_WITH_LOGITS,
+            metric = Metrics.ACCURACY,
+            callback = timeStopping,
```

Comment:
I have one idea here. Are you interested in the ticket about the support of multiple callbacks? At the current time, the API supports only one loss/metric/callback. From my point of view, multiple callbacks could make sense (stopping and logging to the CSV for example)

Reply:
Yeah, sure that's a good idea. Actually, the Python API of Keras always expects a list of callbacks passed to the `fit` method. The same thing could be done here (of course, there is an irrelevant difference here: in KotlinDL, unlike Keras, the callbacks are passed to `compile` method instead of `fit`).

Reply:
Yeah, there is a difference in API due to internal implementation details.